File: test/unit/runtime/print.rb
#
# Copyright 2011 by Salman Ahmad (salman@salmanahmad.com).
# All rights reserved.
#
# Permission is granted for use, copying, modification, distribution,
# and distribution of modified versions of this work as long as the
# above copyright notice is included.
#

require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'test_helper.rb'))

class RuntimeTests::PrintTest < RuntimeTestCase
  
  def test_simple
    # TODO
    return
    
    output = run_code("PRINT 'Hello? Yes, this is Dog.'")
    assert_equal(output, "Hello? Yes, this is Dog.")
    
    output = run_code("INSPECT 'Hello? Yes, this is Dog.'")
    assert_equal(output, "\"Hello? Yes, this is Dog.\"")
  end
  
  
end

File: src/main/java/dog/packages/dog/Print.java
/*
 *
 *  Copyright 2012 by Salman Ahmad (salman@salmanahmad.com).
 *  All rights reserved.
 *
 *  Permission is granted for use, copying, modification, distribution,
 *  and distribution of modified versions of this work as long as the
 *  above copyright notice is included.
 *
 */

package dog.packages.dog;

import dog.lang.Value;
import dog.lang.StringValue;
import dog.lang.Function;
import dog.lang.Signal;
import dog.lang.StackFrame;
import dog.lang.annotation.Symbol;

@Symbol("dog.print:")
public class Print extends Function {

	public int getVariableCount() {
		return 1;
	}

	public Signal resume(StackFrame frame) {
		Value value = frame.variables[0];
		
		if(value instanceof StringValue) {
			System.out.print(value.getValue());
		} else {
			System.out.print(value);
		}
		
		return new Signal(Signal.Type.RETURN);
	}
}

